<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ biomartr</title><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="https://docs.ropensci.org/biomartr/logo.png"><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css"><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript><!-- End Matomo Code --></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">biomartr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.8.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    <a class="dropdown-item" href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    <a class="dropdown-item" href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    <a class="dropdown-item" href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    <a class="dropdown-item" href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/biomartr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="biomart.html">biomart()</a></code> 
        </dt>
        <dd>Main BioMart Query Function</dd>
      </dl><dl><dt>
          
          <code><a href="biomartr-package.html">biomartr-package</a></code> <code><a href="biomartr-package.html">biomartr</a></code> 
        </dt>
        <dd>Genomic Data Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="cachedir.html">cachedir()</a></code> 
        </dt>
        <dd>Get directory to store back end files like kingdom summaries etc</dd>
      </dl><dl><dt>
          
          <code><a href="cachedir_set.html">cachedir_set()</a></code> 
        </dt>
        <dd>Set directory to store back end files like kingdom summaries etc</dd>
      </dl><dl><dt>
          
          <code><a href="check_annotation_biomartr.html">check_annotation_biomartr()</a></code> 
        </dt>
        <dd>Check whether an annotation file contains outlier lines</dd>
      </dl><dl><dt>
          
          <code><a href="download.database.html">download.database()</a></code> 
        </dt>
        <dd>Download a NCBI Database to Your Local Hard Drive</dd>
      </dl><dl><dt>
          
          <code><a href="download.database.all.html">download.database.all()</a></code> 
        </dt>
        <dd>Download all elements of an NCBI databse</dd>
      </dl><dl><dt>
          
          <code><a href="ensembl_divisions.html">ensembl_divisions()</a></code> 
        </dt>
        <dd>List all available ENSEMBL divisions</dd>
      </dl><dl><dt>
          
          <code><a href="get.ensembl.info.html">get.ensembl.info()</a></code> 
        </dt>
        <dd>Helper function to retrieve species information from the ENSEMBL API</dd>
      </dl><dl><dt>
          
          <code><a href="getAssemblyStats.html">getAssemblyStats()</a></code> 
        </dt>
        <dd>Genome Assembly Stats Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getAttributes.html">getAttributes()</a></code> 
        </dt>
        <dd>Retrieve All Available Attributes for a Specific Dataset</dd>
      </dl><dl><dt>
          
          <code><a href="getBio.html">getBio()</a></code> 
        </dt>
        <dd>A wrapper to all bio getters, selected with 'type' argument</dd>
      </dl><dl><dt>
          
          <code><a href="getBioSet.html">getBioSet()</a></code> 
        </dt>
        <dd>Generic Bio data set extractor</dd>
      </dl><dl><dt>
          
          <code><a href="getCDS.html">getCDS()</a></code> 
        </dt>
        <dd>Coding Sequence Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getCDSSet.html">getCDSSet()</a></code> 
        </dt>
        <dd>CDS retrieval of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getCollection.html">getCollection()</a></code> 
        </dt>
        <dd>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats</dd>
      </dl><dl><dt>
          
          <code><a href="getCollectionSet.html">getCollectionSet()</a></code> 
        </dt>
        <dd>Retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getDatasets.html">getDatasets()</a></code> 
        </dt>
        <dd>Retrieve All Available Datasets for a BioMart Database</dd>
      </dl><dl><dt>
          
          <code><a href="getENSEMBL.html">getENSEMBL()</a></code> 
        </dt>
        <dd>Download sequence or annotation from ENSEMBL</dd>
      </dl><dl><dt>
          
          <code><a href="getENSEMBL.Seq.html">getENSEMBL.Seq()</a></code> 
        </dt>
        <dd>Helper function for retrieving biological sequence files from ENSEMBL</dd>
      </dl><dl><dt>
          
          <code><a href="getENSEMBL.gtf.html">getENSEMBL.gtf()</a></code> 
        </dt>
        <dd>Helper function for retrieving gtf files from ENSEMBL</dd>
      </dl><dl><dt>
          
          <code><a href="getENSEMBLGENOMESInfo.html">getENSEMBLGENOMESInfo()</a></code> 
        </dt>
        <dd>Retrieve ENSEMBLGENOMES info file</dd>
      </dl><dl><dt>
          
          <code><a href="getENSEMBLInfo.html">getENSEMBLInfo()</a></code> 
        </dt>
        <dd>Retrieve ENSEMBL info file</dd>
      </dl><dl><dt>
          
          <code><a href="getFilters.html">getFilters()</a></code> 
        </dt>
        <dd>Retrieve All Available Filters for a Specific Dataset</dd>
      </dl><dl><dt>
          
          <code><a href="getGENOMEREPORT.html">getGENOMEREPORT()</a></code> 
        </dt>
        <dd>Retrieve NCBI GENOME_REPORTS file</dd>
      </dl><dl><dt>
          
          <code><a href="getGFF.html">getGFF()</a></code> 
        </dt>
        <dd>Genome Annotation Retrieval (GFF3)</dd>
      </dl><dl><dt>
          
          <code><a href="getGFFSet.html">getGFFSet()</a></code> 
        </dt>
        <dd>GFF retrieval of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getGO.html">getGO()</a></code> 
        </dt>
        <dd>Gene Ontology Query</dd>
      </dl><dl><dt>
          
          <code><a href="getGTF.html">getGTF()</a></code> 
        </dt>
        <dd>Genome Annotation Retrieval (GTF)</dd>
      </dl><dl><dt>
          
          <code><a href="getGenome.html">getGenome()</a></code> 
        </dt>
        <dd>Genome Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getGenomeSet.html">getGenomeSet()</a></code> 
        </dt>
        <dd>Genome Retrieval of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getGroups.html">getGroups()</a></code> 
        </dt>
        <dd>Retrieve available groups for a kingdom of life (only available for NCBI RefSeq and NCBI Genbank)</dd>
      </dl><dl><dt>
          
          <code><a href="getKingdomAssemblySummary.html">getKingdomAssemblySummary()</a></code> 
        </dt>
        <dd>Retrieve and summarise the assembly_summary.txt files from
NCBI for all kingdoms</dd>
      </dl><dl><dt>
          
          <code><a href="getKingdoms.html">getKingdoms()</a></code> 
        </dt>
        <dd>Retrieve available kingdoms of life</dd>
      </dl><dl><dt>
          
          <code><a href="getMarts.html">getMarts()</a></code> 
        </dt>
        <dd>Retrieve information about available Ensembl Biomart databases</dd>
      </dl><dl><dt>
          
          <code><a href="getMetaGenomeAnnotations.html">getMetaGenomeAnnotations()</a></code> 
        </dt>
        <dd>Retrieve annotation *.gff files for metagenomes from NCBI Genbank</dd>
      </dl><dl><dt>
          
          <code><a href="getMetaGenomeSummary.html">getMetaGenomeSummary()</a></code> 
        </dt>
        <dd>Retrieve the assembly_summary.txt file from NCBI genbank metagenomes</dd>
      </dl><dl><dt>
          
          <code><a href="getMetaGenomes.html">getMetaGenomes()</a></code> 
        </dt>
        <dd>Retrieve metagenomes from NCBI Genbank</dd>
      </dl><dl><dt>
          
          <code><a href="getProteome.html">getProteome()</a></code> 
        </dt>
        <dd>Proteome Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getProteomeSet.html">getProteomeSet()</a></code> 
        </dt>
        <dd>Proteome retrieval of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getRNA.html">getRNA()</a></code> 
        </dt>
        <dd>RNA Sequence Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getRNASet.html">getRNASet()</a></code> 
        </dt>
        <dd>RNA Retrieval of multiple species</dd>
      </dl><dl><dt>
          
          <code><a href="getReleases.html">getReleases()</a></code> 
        </dt>
        <dd>Retrieve available database releases or versions
of ENSEMBL</dd>
      </dl><dl><dt>
          
          <code><a href="getRepeatMasker.html">getRepeatMasker()</a></code> 
        </dt>
        <dd>Repeat Masker Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="getSummaryFile.html">getSummaryFile()</a></code> 
        </dt>
        <dd>Helper function to retrieve the assembly_summary.txt file from NCBI</dd>
      </dl><dl><dt>
          
          <code><a href="getUniProtInfo.html">getUniProtInfo()</a></code> 
        </dt>
        <dd>Get uniprot info from organism</dd>
      </dl><dl><dt>
          
          <code><a href="getUniProtSTATS.html">getUniProtSTATS()</a></code> 
        </dt>
        <dd>Retrieve UniProt Database Information File (STATS)</dd>
      </dl><dl><dt>
          
          <code><a href="is.genome.available.html">is.genome.available()</a></code> 
        </dt>
        <dd>Check Genome Availability</dd>
      </dl><dl><dt>
          
          <code><a href="listDatabases.html">listDatabases()</a></code> <code><a href="listDatabases.html">listNCBIDatabases()</a></code> 
        </dt>
        <dd>Retrieve a List of Available NCBI Databases for Download</dd>
      </dl><dl><dt>
          
          <code><a href="listGenomes.html">listGenomes()</a></code> 
        </dt>
        <dd>List All Available Genomes either by kingdom, group, or subgroup</dd>
      </dl><dl><dt>
          
          <code><a href="listGroups.html">listGroups()</a></code> 
        </dt>
        <dd>List number of available genomes in each taxonomic group</dd>
      </dl><dl><dt>
          
          <code><a href="listKingdoms.html">listKingdoms()</a></code> 
        </dt>
        <dd>List number of available genomes in each kingdom of life</dd>
      </dl><dl><dt>
          
          <code><a href="listMetaGenomes.html">listMetaGenomes()</a></code> 
        </dt>
        <dd>List available metagenomes on NCBI Genbank</dd>
      </dl><dl><dt>
          
          <code><a href="meta.retrieval.html">meta.retrieval()</a></code> 
        </dt>
        <dd>Perform Meta-Genome Retrieval</dd>
      </dl><dl><dt>
          
          <code><a href="meta.retrieval.all.html">meta.retrieval.all()</a></code> 
        </dt>
        <dd>Perform Meta-Genome Retrieval of all organisms in all kingdoms of life</dd>
      </dl><dl><dt>
          
          <code><a href="organismAttributes.html">organismAttributes()</a></code> 
        </dt>
        <dd>Retrieve Ensembl Biomart attributes for a query organism</dd>
      </dl><dl><dt>
          
          <code><a href="organismBM.html">organismBM()</a></code> 
        </dt>
        <dd>Retrieve Ensembl Biomart marts and datasets for a query organism</dd>
      </dl><dl><dt>
          
          <code><a href="organismFilters.html">organismFilters()</a></code> 
        </dt>
        <dd>Retrieve Ensembl Biomart filters for a query organism</dd>
      </dl><dl><dt>
          
          <code><a href="read_assemblystats.html">read_assemblystats()</a></code> 
        </dt>
        <dd>Import Genome Assembly Stats File</dd>
      </dl><dl><dt>
          
          <code><a href="read_cds.html">read_cds()</a></code> 
        </dt>
        <dd>Import CDS as Biostrings or data.table object</dd>
      </dl><dl><dt>
          
          <code><a href="read_genome.html">read_genome()</a></code> 
        </dt>
        <dd>Import Genome Assembly as Biostrings or data.table object</dd>
      </dl><dl><dt>
          
          <code><a href="read_gff.html">read_gff()</a></code> 
        </dt>
        <dd>Import GFF File</dd>
      </dl><dl><dt>
          
          <code><a href="read_proteome.html">read_proteome()</a></code> 
        </dt>
        <dd>Import Proteome as Biostrings or data.table object</dd>
      </dl><dl><dt>
          
          <code><a href="read_rm.html">read_rm()</a></code> 
        </dt>
        <dd>Import Repeat Masker output file</dd>
      </dl><dl><dt>
          
          <code><a href="read_rna.html">read_rna()</a></code> 
        </dt>
        <dd>Import RNA as Biostrings or data.table object</dd>
      </dl><dl><dt>
          
          <code><a href="refseqOrganisms.html">refseqOrganisms()</a></code> 
        </dt>
        <dd>Retrieve All Organism Names Stored on refseq</dd>
      </dl><dl><dt>
          
          <code><a href="summary_cds.html">summary_cds()</a></code> 
        </dt>
        <dd>Retrieve summary statistics for a coding sequence (CDS) file</dd>
      </dl><dl><dt>
          
          <code><a href="summary_genome.html">summary_genome()</a></code> 
        </dt>
        <dd>Retrieve summary statistics for a genome assembly file</dd>
      </dl></div>
  </main></div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg"></div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul><h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul></div>
                    <div class="col-md-3 col-sm-4">
                        <ul><h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul></div>
                    <div class="col-md-2 col-sm-4">
                        <ul><h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul></div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer></div>

  

  

  </body></html>

